File: auction/src/services/productService.js
import apiClient from '../api/apiClient';

const productService = {
    getAllProducts: () => apiClient.get('/product'),
    addProduct: (productData) => apiClient.post('/product', productData),
    getProductById: (productId) => apiClient.get(`/product/${productId}`),
    updateProduct: (productId, productData) =>
        apiClient.put('/product', productData, { params: { id: productId } })
};

export default productService;

File: AuctionWebsite/ProductService/src/main/java/com/product/service/ProductService.java
package com.product.service;

import java.util.List;

import com.product.dto.ProductDTO;
import com.product.entity.Product;

public interface ProductService {

	List<Product> getAllProducts();

	Product getProductById(int id);

	Product saveProduct(Product product);

	String deleteProduct(int id);

	Product updateProduct(int id, Product product);

	ProductDTO getByProductId(int id);

	void setStatus(int id, String status);

}
